### PR TITLE
[BLOCKS] Add a max entry on contentset

### PR DESCRIPTION
**Pull request by cmarianhodis**
Related issue: https://github.com/backbee/backbee-standard/issues/568

### CODE REVIEW

File: src/tb/apps/content/components/DndManager.js [hunk 1]
```diff
@@ -253,7 +253,13 @@ define(
                             }
 
                             if (children.length > 0) {
-                                this.putDropZoneAroundContentSetChildren(children, dropzoneEl, currentContentId, contentSet);
+                                if (contentSet.maxEntry !== '') {
```
Comment (fkroockmann) [inline src/tb/apps/content/components/DndManager.js]: can you do that with on line like: if (contentSet.maxEntry && children.length < contentSet.maxEntry) {

File: src/tb/apps/content/models/ContentSet.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ define(
         var ContentSet = new JS.Class(AbstractContent, {
 
             identifierDataAttribute: 'bb-identifier',
+            maxEntryDataAttribute: 'bb-maxentry',
```
Comment (fkroockmann) [inline src/tb/apps/content/models/ContentSet.js]: put a default value here like : this.maxEntry: null,

File: src/tb/apps/content/plugins/contentsetplus/main.js [hunk 3]
```diff
@@ -151,7 +151,16 @@ define(
...
+                var content = this.getCurrentContent(),
+                    check = false;
+                if (content.isAContentSet() && content.maxEntry !== '') {
+                    if (content.getNodeChildren().length < content.maxEntry) {
+                        check = true;
+                    }
+                } else if (content.isAContentSet() && content.maxEntry === '') {
+                    check = true;
+                }
+                return check;
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentsetplus/main.js]: add space between } and return plz

**Comment by fkroockmann** [comment]
Can you replace all content.maxEntry !== ' ' by content.maxEntry

**Comment by cmarianhodis** [comment]
I made the requested adjustments

File: src/tb/apps/content/plugins/contentselector/main.js [hunk 4]
```diff
@@ -57,7 +57,17 @@ define(
...
 
             canApplyOnContext: function () {
-                return this.getCurrentContent().isAContentSet();
+                var content = this.getCurrentContent(),
+                    check = false;
+                if (content.isAContentSet() && content.maxEntry) {
+                    if (content.getNodeChildren().length < content.maxEntry) {
+                        check = true;
+                    }
+                } else if (content.isAContentSet() && content.maxEntry === '') {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentselector/main.js]: you forgot here

File: src/tb/apps/content/plugins/contentselector/main.js [hunk 5]
```diff
@@ -57,7 +57,17 @@ define(
             },
 
             canApplyOnContext: function () {
-                return this.getCurrentContent().isAContentSet();
+                var content = this.getCurrentContent(),
+                    check = false;
+                if (content.isAContentSet() && content.maxEntry) {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentselector/main.js]: you can do it like that

    if (content.isAContentSet()) {
        if (content.maxEntry) {
            if (content.getNodeChildren().length < content.maxEntry) {
                check = true;
            }
        } else {
            check = true;
        }
    }

File: src/tb/apps/content/plugins/contentsetplus/main.js [hunk 6]
```diff
@@ -151,7 +151,17 @@ define(
              * @returns {Boolean}
              */
             canApplyOnContext: function () {
-                return this.getCurrentContent().isAContentSet();
+                var content = this.getCurrentContent(),
+                    check = false;
+                if (content.isAContentSet() && content.maxEntry) {
```
Comment (fkroockmann) [inline src/tb/apps/content/plugins/contentsetplus/main.js]: same here

File: src/tb/apps/content/models/ContentSet.js [hunk 7]
```diff
@@ -43,6 +45,12 @@ define(
              */
             initialize: function (config) {
                 this.callSuper(config);
+
+                if (this.jQueryObject instanceof jQuery) {
+                    if (this.jQueryObject.data(this.maxEntryDataAttribute) !== undefined) {
```
Comment (fkroockmann) [inline src/tb/apps/content/models/ContentSet.js]: can you just if(this.jQueryObject.data(this.maxEntryDataAttribute))

**Pull request merged**